src/main/kotlin/uk/gov/dluhc/printapi/messaging/RemoveCertificateListener.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.dluhc.printapi.messaging
+
+import io.awspring.cloud.messaging.listener.annotation.SqsListener
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.messaging.models.RemoveCertificateMessage
+import uk.gov.dluhc.printapi.service.CertificateDataRetentionService
+import javax.validation.Valid
+
+/**
+ * Implementation of [MessageListener] to handle [RemoveCertificateMessage] messages
+ */
+@Component
+class RemoveCertificateListener(
+    val certificateDataRetentionService: CertificateDataRetentionService
+) : MessageListener<RemoveCertificateMessage> {
+
+    @SqsListener("\${sqs.remove-certificate-queue-name}")
+    override fun handleMessage(@Valid @Payload payload: RemoveCertificateMessage) {
+        certificateDataRetentionService.removeFinalRetentionPeriodData(payload)
```

Comment:
It will be nicer to add a logger info stmt

Reply:
I considered adding a debug one. Trouble is there could be over 1 million rows on 1st July each year, so not sure it would be that useful :)